rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala

```diff
@@ -543,6 +543,39 @@ class ProcMatcherSpec extends FlatSpec with Matchers {
       ProcNormalizeMatcher.normalizeMatch(pMatch, inputs)
     }
   }
+  "PMatch" should "Handle a match inside a for pattern" in {
+    // for (@(match (x | y) { case 47 => Nil }) <- @Nil) { Nil }
+
+    val listCases = new ListCase()
+    listCases.add(new CaseImpl(new PGround(new GroundInt(47)), new PNil()))
+    val pMatch       = new PMatch(new PPar(new PVar("x"), new PVar("y")), listCases)
+    val listBindings = new ListName()
+    listBindings.add(new NameQuote(pMatch))
+    val listLinearBinds = new ListLinearBind()
+    listLinearBinds.add(new LinearBindImpl(listBindings, new NameQuote(new PNil())))
+    val linearSimple = new LinearSimple(listLinearBinds)
+    val receipt      = new ReceiptLinear(linearSimple)
+    var input        = new PInput(receipt, new PNil())
+
+    val result    = ProcNormalizeMatcher.normalizeMatch(input, inputs)
+    val bindCount = 2
```

Comment:
Just wanted to put particular attention to this bindCount. I guess the alternative that is possible is 0.

Reply:
Nope, it should be two. The match target has 2 free variables, and so the whole match expression does as well.

Reply:
Ok good to confirm.